packages/push/src/instruction/printing/mod.rs

```diff
@@ -0,0 +1,387 @@
+mod char;
+mod string;
+
+use std::{fmt::Display, io::Write, marker::PhantomData};
+
+pub use char::{PrintChar, PrintNewline, PrintPeriod, PrintSpace};
+pub use string::PrintString;
+
+use super::{NumOpens, instruction_error::PushInstructionError};
+use crate::{
+    error::{Error, InstructionResult},
+    instruction::Instruction,
+    push_vm::{HasStack, push_io::HasStdout},
+};
+
+/// An instruction that "prints" the top value of the stack
+/// of type `T` to an internal buffer in the state.
+///
+/// # Inputs
+///
+/// The `Print<T>` instruction takes the following inputs:
+///    - `T` stack
+///      - One value
+///
+/// # Behavior
+///
+/// The `Print<T>` instruction "prints" the top value of
+/// the stack of type `T` to an internal buffer in the state.
+///
+/// This does _not_ put a newline after the printed value;
+/// see `PrintLn` for that behavior.
```

Comment:
I would somehow note that this uses rust's `Display` implementation on the stack type to do the conversion to a string to print. We could reasonably have a PrintDebug as well, although I don't know if that would be that useful.

Reply:
I added it up in the "introduction" as a new 2nd paragraph.

Reply:
We could have a `PrintDebug`, but I don't think we need it _now_. Maybe create an issue for it, but I'm not sure how important it is.